### PR TITLE
RGWs: Make multiple instances do not interfere with MGRs

### LINKED ISSUE #1483: RGW: Ceph managers enter in a crashloop.
**Bug Report**

Ceph MGRs enter in a crashloop all of a sudden when the replica count if an objectstore is bigger than 1

How to reproduce it (minimal and precise):
Just set the replicacount to any natural number bigger than 1. The issue is that the rados gateways run all with the same name, and do not have any random suffix to them.

### PR DESCRIPTION
**Pull request by dgonzalezruiz**
Description of your changes: This makes the RGWs commandline tool use the HOSTNAME environment variable, which has a randomized suffix, so the need of different name for different gateways is true.

Resolves #1483

### CODE REVIEW

File: cmd/rook/rgw.go [hunk 1]
```diff
@@ -40,9 +41,9 @@ var (
 )
 
 func init() {
-	rgwCmd.Flags().StringVar(&rgwName, "rgw-name", "", "name of the object store")
+	rgwCmd.Flags().StringVar(&rgwName, "rgw-name", "", "name of the object store. Defaults to the pod hostname.")
```
Comment (galexrt) [inline cmd/rook/rgw.go]: Doesn't this `Defaults ...` needs to be moved to `rgwHost` two lines below?

File: tests/integration/base_object_test.go [hunk 2]
```diff
@@ -121,7 +121,11 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
 	require.Equal(s.T(), len(usersBeforeDelete)-1, len(usersAfterDelete), "Make sure user list count is reducd by 1")
 	logger.Infof("Object store user deleted successfully")
 
-	logger.Infof("Step 8: Delete Object Store")
+	logger.Infof("Step 9: Check that MGRs are not in a crashloop")
```
Comment (travisn) [inline tests/integration/base_object_test.go]: what happened to step 8? Want to change the second step 7 while you're at it?

**Comment by dgonzalezruiz** [comment]
@travisn I wasn't actually able to reproduce this issue privately with the minikube environment. Maybe higher RGW activity is required. Do you suggest anything? I would also love to be able to test this properly in the smoke tests.
MGRs seem to be stable there. In my private cluster, this does totally fix the issue. This is the right way of configuring the RGWs according to this http://lists.ceph.com/pipermail/ceph-users-ceph.com/2017-September/020712.html

**Comment by travisn** [comment]
@dgonzalezruiz sounds good on the investigation and that you were able to verify the fix. The changes look good. Could you just squash the commits? Then we'll be ready to merge.

**Comment by travisn** [comment]
@galexrt @dgonzalezruiz We are working on resolving an issue with GCE, unfortunately it isn't resolved yet

**Pull request merged**